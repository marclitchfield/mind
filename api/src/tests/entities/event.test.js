import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe.only('Event', () => {
  applySetup();

  test('createInSpace', () => {
    return verifyMutations('Event', 'createInSpace', {
      mutations: [
        ({space}) => gql`mutation { 
          Event { event:createInSpace(spaceId:"${space.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } }
        }`
      ],
      query: ({space}) => gql`query { 
        Space(id:"${space.id}") { 
          events { title spaces { title } }
        } 
      }`
    });
  });

  test('createForConcept', () => {
    return verifyMutations('Event', 'createForConcept', {
      mutations: [
        ({space}) => gql`mutation { Concept { concept:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({concept}) => gql`mutation { Event { event:createForConcept(conceptId:"${concept.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          concepts { title events { title } }
        } 
      }`
    });
  });

  test('createAtLocation', () => {
    return verifyMutations('Event', 'createAtLocation', {
      mutations: [
        ({space}) => gql`mutation { Location { location:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({location}) => gql`mutation { Event { event:createAtLocation(locationId:"${location.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          locations { title timeline { title }}
        } 
      }`
    });
  });

  test('createForCollection', () => {
    return verifyMutations('Event', 'createForCollection', {
      mutations: [
        ({space}) => gql`mutation { Concept { concept:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({concept}) => gql`mutation { Collection { collection:createInstance(classConceptId:"${concept.id}" input:{title:"test"}) { id } } }`,
        ({collection}) => gql`mutation { Event { event:createForCollection(collectionId:"${collection.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          collections { title timeline { title } }
        } 
      }`
    });
  });

  test('createForItem', () => {
    return verifyMutations('Event', 'createForItem', {
      mutations: [
        ({space}) => gql`mutation { Concept { concept:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({concept}) => gql`mutation { Item { item:createInstance(classConceptId:"${concept.id}" input:{title:"test"}) { id } } }`,
        ({item}) => gql`mutation { Event { event:createForItem(itemId:"${item.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          items { title timeline { title } }
        } 
      }`
    });
  });

  test('createForIdea', () => {
    return verifyMutations('Event', 'createForIdea', {
      mutations: [
        ({space}) => gql`mutation { Idea { idea:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({idea}) => gql`mutation { Event { event:createForIdea(ideaId:"${idea.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          ideas { title events { title }}
        } 
      }`
    });
  });

  test('createForPerson', () => {
    return verifyMutations('Event', 'createForPerson', {
      mutations: [
        ({space}) => gql`mutation { Person { person:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({person}) => gql`mutation { Event { event:createForPerson(personId:"${person.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          people { title timeline { title }}
        } 
      }`
    });
  });

  test('addLocation', () => {
    return verifyMutations('Event', 'addLocation', {
      mutations: [
        ({space}) => gql`mutation { Location { location:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:createInSpace(spaceId:"${space.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`,
        ({location, event}) => gql`mutation { Event { addLocation(id:"${event.id}" locationId:"${location.id}") } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          locations { title timeline { title }}
        } 
      }`
    });
  });

  test('removeLocation', () => {
    return verifyMutations('Event', 'removeLocation', {
      mutations: [
        ({space}) => gql`mutation { Location { location:createInSpace(spaceId:"${space.id}" input:{title:"test"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:createInSpace(spaceId:"${space.id}" input:{title: "Test", datetime: "2011-12-12"}) { id } } }`,
        ({location, event}) => gql`mutation { Event { addLocation(id:"${event.id}" locationId:"${location.id}") } }`,
        ({location, event}) => gql`mutation { Event { removeLocation(id:"${event.id}" locationId:"${location.id}") } }`
      ],
      query: ({event}) => gql`query { 
        Event(id:"${event.id}") { 
          locations { title timeline { title }}
        } 
      }`
    });
  });
});