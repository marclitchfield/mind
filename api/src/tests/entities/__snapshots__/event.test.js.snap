// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event addLocation 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "locations": Array [
      Object {
        "__typename": "Location",
        "timeline": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createAtLocation 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "locations": Array [
      Object {
        "__typename": "Location",
        "timeline": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createForCollection 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "collections": Array [
      Object {
        "__typename": "Collection",
        "timeline": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createForConcept 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "concepts": Array [
      Object {
        "__typename": "Concept",
        "events": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createForIdea 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "ideas": Array [
      Object {
        "__typename": "Idea",
        "events": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createForItem 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "items": Array [
      Object {
        "__typename": "Item",
        "timeline": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createForPerson 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "people": Array [
      Object {
        "__typename": "Person",
        "timeline": Array [
          Object {
            "__typename": "Event",
            "title": "Test",
          },
        ],
        "title": "test",
      },
    ],
  },
}
`;

exports[`Event createInSpace 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "events": Array [
      Object {
        "__typename": "Event",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "createInSpace",
          },
        ],
        "title": "Test",
      },
    ],
  },
}
`;

exports[`Event removeLocation 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "locations": Array [],
  },
}
`;
